### PR TITLE
modules/k8s: Tunnel can have a custom logger

### PR DESCRIPTION
**Pull request by ishustava**
With #510, there is now an ability to provide your own custom logger, which is awesome! However, the `Tunnel` struct used for port-forwading in the `k8s` module doesn't allow its logger to be customized. This PR proposes to use a custom logger in the for `Tunnel`.

### CODE REVIEW

**Comment by brikis98** [comment]
I'm not familiar with this code, so assigning to @yorinasub17 to take a look when he's back.

File: modules/k8s/tunnel.go [hunk 1]
```diff
@@ -66,20 +66,34 @@ type Tunnel struct {
...
+// NewTunnelWithLogger will create a new Tunnel struct with the provided logger.
+// Note that if you use 0 for the local port, an open port on the host system
+// will be selected automatically, and the Tunnel struct will be updated with the selected port.
+func NewTunnelWithLogger(
+	kubectlOptions *KubectlOptions,
+	resourceType KubeResourceType,
+	resourceName string,
+	local int,
+	remote int,
+	logger logger.TestLogger) *Tunnel {
```
Comment (yorinasub17) [inline modules/k8s/tunnel.go:88]: NIT: The style we use is:
```suggestion
	logger logger.TestLogger,
) *Tunnel {
```

**Review by yorinasub17 — COMMENTED** [review]
LGTM! I'll kick off a test build now.

**Comment by yorinasub17** [comment]
Looks like the test failed, but the failures are unrelated to this change so we can get this merged once you address the style issue and update to include the latest changes!

File: modules/k8s/tunnel.go (hunk 1 above)
Comment (ishustava) [reply modules/k8s/tunnel.go:88]: makes sense!

**Review by zackproser — APPROVED** [review]
LGTM :+1:

**Comment by yorinasub17** [comment]
Thanks for the contribution! This is good to go, so will merge and release now.

**Pull request merged**